src/ui/public/config/config.js

```diff
@@ -63,18 +63,20 @@ any custom setting configuration watchers for "${key}" may fix this issue.`);
     const newVal = key in defaults && defaults[key].defaultValue === value ? null : value;
     const unchanged = oldVal === newVal;
     if (unchanged) {
-      return Promise.resolve();
+      return Promise.resolve(true);
     }
     const initialVal = declared ? config.get(key) : undefined;
     localUpdate(key, newVal, initialVal);
 
     return delayedUpdate(key, newVal)
       .then(updatedSettings => {
         settings = mergeSettings(defaults, updatedSettings);
+        return true;
       })
       .catch(reason => {
         localUpdate(key, initialVal, config.get(key));
         notify.error(reason);
+        return false;
```

Comment:
is there a way to have this return value covered in a test?

Reply:
Not without exposing the `change` method and allowing the `delayedUpdate` method to be overridden, or adding the override to `change` _and_ `set`, which is undesirable.

Reply:
I decided it was worth the effort, so I exposed `change` as `_change` and added tests for `true` and `false` scenarios.